### DISCUSSION TITLE
C# 함수 호출 가능한가요?

### DISCUSSION THREAD

**Discussion by cinephile231** [naverz/zepeto-studio-kor#705] [5. Q&A]
C# 함수 호출 가능한가요?
이게 제페토에서 사용하는 스크립트가 정확히 자바스크립트는 아닌 거죠?
방법 좀 알려주셔요..

**Comment by cinephile231** [comment]
SendMessage로 하니까 되기는 하는데.. 쩝
더 멋진 방법을 알고 싶네요.

C# script

```
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class PlayerCtrl : MonoBehaviour
{
    public void LocalPlayerSetting()
    {
        Debug.Log("LocalPlayerSetting");
    }
}
```

Zepeto Script
`this.SendMessage("LocalPlayerSetting");`

같은 오브젝트 안에 있어야 함수를 찾습니다...

**Comment by Joyounwoo** [comment]
제페토스크립트에서 사용하는 언어는 TypeScript 입니다.
하지만 ts는 컴파일 언어로 js로 컴파일되어 사용되고 최종적으로 빌드에 올라가는 것은 js입니다.

C# 스크립트 넣으셔도 실제 제페토 앱에서는 동작하지 않는 것으로 알고 있습니다.
가능한 제페토스크립트로 기능을 구현하시는 것이 타당해 보입니다.
